tests/examples/testWorkflowExamples.cpp

```diff
@@ -0,0 +1,105 @@
+
+#include <catch2/catch_test_macros.hpp>
+
+#include "BaseIO.hpp"
+#include "Channel.hpp"
+#include "Types.hpp"
+#include "Utils.hpp"
+#include "hdf5/HDF5IO.hpp"
+#include "nwb/NWBFile.hpp"
+#include "nwb/RecordingContainers.hpp"
+#include "nwb/file/ElectrodeTable.hpp"
+#include "testUtils.hpp"
+
+using namespace AQNWB;
+
+TEST_CASE("workflowExamples")
+{
+  SECTION("write workflow")
+  {
+    // 0. setup mock data
+    SizeType numChannels = 4;
+    SizeType numSamples = 300;
+    SizeType samplesRecorded = 0;
+    SizeType bufferSize = numSamples / 10;
+    std::vector<float> dataBuffer(bufferSize);
+    std::vector<double> timestampsBuffer(bufferSize);
+
+    std::vector<Types::ChannelVector> mockRecordingArrays =
+        getMockChannelArrays();
+    std::vector<std::vector<float>> mockData =
+        getMockData2D(numSamples, numChannels);
+    std::vector<double> mockTimestamps = getMockTimestamps(numSamples);
+
+    std::string path = getTestFilePath("exampleRecording.nwb");
+    // [example_workflow_io_snippet]
+    std::shared_ptr<BaseIO> io = createIO("HDF5", path);
+    // [example_workflow_io_snippet]
+
+    // [example_workflow_recording_containers_snippet]
+    std::unique_ptr<NWB::RecordingContainers> recordingContainers =
+        std::make_unique<NWB::RecordingContainers>();
+    // [example_workflow_recording_containers_snippet]
+
+    // [example_workflow_nwbfile_snippet]
+    std::unique_ptr<NWB::NWBFile> nwbfile =
+        std::make_unique<NWB::NWBFile>(generateUuid(), io);
+    nwbfile->initialize();
+    // [example_workflow_nwbfile_snippet]
+
+    // [example_workflow_datasets_snippet]
+    nwbfile->createElectricalSeries(
```

Comment:
How does the user know what the index of the `ElectricalSeries` is? I.e., how does the user know how to access the `ElectricalSeries`? Is this currently implicit, i.e., the user needs to remember the order in which they added objects to the `recordingContainers` object? 

If so, I'd suggest to either: 1) change `createElecticalSeries` to return the index of the object that was added, or 2) add a line here along the lines of ``SizeType electricalSeriesIndex = recordingContainers.get().size() - 1`` along with some text in the workflow docs to indicate that : a) new objects are always appended at the end in `RecordingContainers` and b) that this is how we can get and remember the index needed to retrieve the object later on. Either option seems fine, but it will be useful to document how we can get the index we should use to access a particular ``Container`` that was added to ``RecordingContainers``

Reply:
I added `containerIndexes` as an optional input that will let users keep track of which index has been created by `createElectricalSeries`.

I think this approach will keep it more consistent that functions that might modify the NWBFile like `createElectricalSeries` return their status (e.g. returning a failure if the `io->canModifyObjects()` is false). 

`createElectricalSeries` can also add multiple containers (depending on the size of `recordingArrays`) to the `recordingContainers` object so I think having it within the function might also be easier for the user rather than keeping track of the size before and after `createElectricalSeries` is called.